### PR TITLE
Implement MenuBar

### LINKED ISSUE #2: Implement Java API
Investigate the `ContextMenu` API and implement the correlated one for the `Menu`.
Initial API plan:
- Adding items
- Removing items
- Layout setting

### LINKED ISSUE #3: Add unit tests for the API

### LINKED ISSUE #4: Add integration tests for the API
- Add Integration Tests
- Add Testbench element + API

### LINKED ISSUE #5: Create demos & documentation

### PR DESCRIPTION
**Pull request by pekam**
Closes #2, closes #3, closes #4

### CODE REVIEW

File: vaadin-menu-bar-flow/src/test/java/com/vaadin/flow/component/menubar/tests/MenuBarTest.java [hunk 1]
```diff
@@ -15,6 +15,81 @@
...
+    public void openOnHover_falseByDefault() {
+        Assert.assertFalse(menuBar.isOpenOnHover());
+    }
+
+    @Test
+    public void setOpenOnHover_isOpenOnHover() {
+        menuBar.setOpenOnHover(true);
+        Assert.assertTrue(menuBar.isOpenOnHover());
+    }
+
```
Comment (yuriy-fix) [inline vaadin-menu-bar-flow/src/test/java/com/vaadin/flow/component/menubar/tests/MenuBarTest.java]: Should we add test for `resetContent` after initial rendering?

File: vaadin-menu-bar-flow-integration-tests/src/test/java/com/vaadin/flow/component/menubar/tests/MenuBarPageIT.java [hunk 2]
```diff
@@ -15,23 +15,238 @@
...
+        verifyClosed();
+        assertMessage("false");
+
+        openSubSubMenu();
+        checkableItem = getOverlayMenuItems(getAllOverlays().get(1)).get(1);
+        Assert.assertFalse(checkableItem.hasAttribute("menu-item-checked"));
+    }
+
+    @Test
+    public void addRootItem_newRootItemRendered() {
```
[the remark this answers is not in the document]
Comment (pekam) [reply vaadin-menu-bar-flow-integration-tests/src/test/java/com/vaadin/flow/component/menubar/tests/MenuBarPageIT.java]: We already have tests to both adding to sub-menu and adding items on root-level. I think those cover it enough.

File: vaadin-menu-bar-flow/src/main/java/com/vaadin/flow/component/menubar/MenuBar.java [hunk 3]
```diff
@@ -15,20 +15,227 @@
...
+                (menu, contentReset) -> new MenuBarRootItem(contentReset),
+                MenuItem.class, null);
+        addAttachListener(event -> resetContent());
+    }
+
+    /**
+     * Adds a new item component with the given text content to the root level
+     * of this menu bar.
+     * <p>
+     * The root level items are normally placed inside buttons. When the buttons
```
[the remark this answers is not in the document]
Comment (pekam) [reply vaadin-menu-bar-flow/src/main/java/com/vaadin/flow/component/menubar/MenuBar.java]: JavaDocs improved

File: vaadin-menu-bar-flow-integration-tests/src/main/java/com/vaadin/flow/component/menubar/tests/MenuBarTestPage.java [hunk 4]
```diff
@@ -25,5 +29,48 @@
...
+        add(message);
+
+        MenuItem item1 = menuBar.addItem("item 1");
+
+        MenuItem item2 = menuBar.addItem(new Paragraph("item 2"),
+                e -> message.setText(message.getText() + "clicked item 2"));
+
+        MenuItem subItem1 = item1.getSubMenu().addItem("sub item 1");
+        MenuItem subItem2 = item1.getSubMenu()
+                .addItem(new Paragraph("sub item 2"));
```
[the remark this answers is not in the document]
Comment (pekam) [reply vaadin-menu-bar-flow-integration-tests/src/main/java/com/vaadin/flow/component/menubar/tests/MenuBarTestPage.java]: Done

File: vaadin-menu-bar-flow-demo/src/main/java/com/vaadin/flow/component/menubar/demo/MenuBarView.java [hunk 5]
```diff
@@ -27,8 +32,42 @@
 
     @Override
     public void initView() {
+        createBasicDemo();
+    }
+
+    private void createBasicDemo() {
```
[the remark this answers is not in the document]
Comment (pekam) [reply vaadin-menu-bar-flow-demo/src/main/java/com/vaadin/flow/component/menubar/demo/MenuBarView.java]: I'll remove the closing statement and leave the demo improvements for future work

File: vaadin-menu-bar-flow-demo/pom.xml [hunk 6]
```diff
@@ -49,6 +49,12 @@
             <version>${project.version}</version>
         </dependency>
 
+        <dependency>
+            <groupId>com.vaadin</groupId>
+            <artifactId>vaadin-icons-flow</artifactId>
+            <version>1.3.1</version>
+        </dependency>
+
```
[the remark this answers is not in the document]
Comment (pekam) [reply vaadin-menu-bar-flow-demo/pom.xml]: Done

**Pull request merged**